### PR TITLE
refactor(web): matched gesture-component abstraction 🐵

### PR DESCRIPTION
**Pull request by jahorton**
This PR's primary changes seek to facilitate app/webview's delegated longpress behavior as used by our Android app.

While working on #9455, I noticed that a little abstraction would be needed in order to be compatible with such longpresses.  In such cases, the "subkey menu" aspect of a longpress would not have any corresponding `GestureMatcher` instance... and up until now, I've written further implementation assuming that one would exist.

Long story short, this led to the `PredecessorMatch` interface defined herein; it's reasonable to construct a version of this on behalf of a delegated longpress/subkey menu.  Doing so allows us to treat the result of that delegation similarly to a gesture component recognized by the gesture-engine itself, mitigating the need for further code divergence in support of Android-app longpresses.

While I was at it, I figured that one of the Matcher properties could use a better, clearer name... so I made that change.

Both sets of changes forced a bit of unit-test maintenance, but it was fortunately pretty small-scale - and it even makes one aspect of certain test-setups clearer, which was a nice bonus.

@keymanapp-test-bot skip

### CODE REVIEW

File: common/web/gesture-recognizer/src/engine/headless/gestures/matchers/gestureMatcher.ts [hunk 1]
```diff
@@ -5,22 +5,36 @@ import { GestureModel, GestureResolution, GestureResolutionSpec, RejectionDefaul
 import { ManagedPromise, TimeoutPromise } from "@keymanapp/web-utils";
 import { FulfillmentCause, PathMatcher } from "./pathMatcher.js";
 
+export interface PredecessorMatch<Type> {
```
Comment (mcdurdin) [inline common/web/gesture-recognizer/src/engine/headless/gestures/matchers/gestureMatcher.ts:8]: Can you include a comment on what this is for? It's not entirely clear just from reading the code grin

File: common/web/gesture-recognizer/src/engine/headless/gestures/matchers/gestureMatcher.ts [hunk 2]
```diff
@@ -5,22 +5,36 @@ import { GestureModel, GestureResolution, GestureResolutionSpec, RejectionDefaul
...
-  matched: boolean,
-  action: GestureResolution<Type>
+  readonly matched: boolean,
+  readonly action: GestureResolution<Type>
 }
 
-export interface MatchResultSpec {
-  matched: boolean,
-  action: GestureResolutionSpec
+export interface MatchResultSpec<Type> {
```
Comment (mcdurdin) [inline common/web/gesture-recognizer/src/engine/headless/gestures/matchers/gestureMatcher.ts:22]: Why does this need `<Type>` given `<Type>` is not used anywhere in the interface?

**Review by mcdurdin — APPROVED** [review]
LGTM

**Review by jahorton — COMMENTED** [review]
Meant to post this yesterday, but apparently never actually submitted it.

**Pull request merged**